packages/web-components/src/components/va-checkbox/va-checkbox.tsx

```diff
@@ -230,14 +256,15 @@ export class VaCheckbox {
             aria-describedby={ariaDescribedbyIds}
             aria-invalid={error ? 'true' : 'false'}
             disabled={disabled}
+            data-indeterminate={indeterminate && !checked}
             onChange={this.handleChange}
           />
           <label
             htmlFor="checkbox-element"
             class="usa-checkbox__label"
             part="label"
             role="checkbox"
-            aria-checked={ariaChecked}
+            aria-checked={indeterminate && !checked ? 'mixed' : ariaChecked}
```

Comment:
I'm adding this because otherwise the `aria-checked` state would be either `true` or `false`. It looks like it also accepts the "mixed" value [according to MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked). 

But setting `indeterminate` on the input element seems to be what is making the screenreader announcement.